lib/streamlit/components/types/base_component_registry.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Protocol
+
+from streamlit.components.types.base_custom_component import BaseCustomComponent
+
+
+class BaseComponentRegistry(Protocol):
+    """Interface for ComponentRegistries."""
+
+    @abstractmethod
+    def register_component(self, component: BaseCustomComponent) -> None:
+        """Register a CustomComponent.
+
+        Parameters
+        ----------
+        component : CustomComponent
+            The component to register.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_component_path(self, name: str) -> str | None:
```

Comment:
nit: One aspect to consider: we could also just provide a single `get_component(name: str) -> BaseCustomComponent | None` method instead of `get_component_path` and `get_module_name`. I don't really have any opinion if that makes sense or not, but could make the interface a bit more flexible since we don't have to add a new methods for new relevant properties. And the `BaseCustomComponent` is anyways an integrated part of the interface because of `register_component` and `get_components `

Reply:
good point, I have added that method but kept the others for now to not increase the refactoring work even more. We can think about whether we remove the others, but at least adding this method provides the flexibility you mentioned.